### DISCUSSION TITLE
[CodeQL] How do you exclude some files / directories in the creation of a database?

### DISCUSSION THREAD

**Discussion by JasperSurmont** [github/securitylab#738] [Q&A]
A large part of my Go project is scaffolded (boilerplate). Is there a way, when creating the database, to exclude these prebuilt files and only include certain (self-written) files?

**Comment by jketema** [comment]
Hi @JasperSurmont. Thanks for your question. It's unfortunately not possible to exclude some of the files, as CodeQL needs a somewhat complete view of your codebase in order to reason able the code. There are several options here:
* If your code has several build targets, you may just want to build those that include the code you're interested in analyzing.
* If your eventual aim is to filter alerts in code scanning, then we recommend dismissing the alerts in the code scanning UI. However there's also the [`filter-sarif` action](https://github.com/advanced-security/filter-sarif) and [`dismiss alerts` action](https://github.com/advanced-security/dismiss-alerts) that may help here.
